packages/core/__tests__/interpreter/cancellation.js

```diff
@@ -760,3 +760,31 @@ test('should bubble an exception thrown during cancellation', () => {
 
   return expect(middleware.run(worker).toPromise()).rejects.toBe(expectedError)
 })
+
+test('task should end in cancelled state when receiving a cancellation from the child', () => {
+  const middleware = sagaMiddleware()
+  createStore(() => ({}), {}, applyMiddleware(middleware))
+
+  function* child() {
+    yield io.delay(0)
+    yield io.cancel()
+  }
+
+  function* parent() {
+    yield io.fork(child)
+  }
+
+  let task
+  function* worker() {
+    task = yield io.fork(parent)
+  }
+
+  return middleware
+    .run(worker)
+    .toPromise()
+    .then(() => {
+      expect(task.isCancelled()).toBe(true)
```

Comment:
This is expected to be false. Cancellation won't bubble from child to parent.  Uncaught errors bubbles from child to parent, and cancellation propagates from parent to children.

Reply:
Right, this was a quick draft and I have not finished it (and obviously made a mistake here).